packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -117,91 +113,96 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
         document.addEventListener("keyup", this._handleKeyup);
     }
 
     removeEventListeners() {
-        document.removeEventListener("click", this._handleInteract);
-        document.removeEventListener("touch", this._handleInteract);
+        document.removeEventListener("mouseup", this._handleInteract);
+        document.removeEventListener("touchend", this._handleInteract);
         document.removeEventListener("keyup", this._handleKeyup);
     }
 
     _handleInteract = (event: {target: any}) => {
-        if (this.props.open) {
-            this.props.onDropdownChanged(false);
+        const target: Node = event.target;
+        if (this.props.open && this.node && !this.node.contains(target)) {
+            this.props.onOpenChanged(false);
         }
     };
 
     _handleKeyup = (event: KeyboardEvent) => {
         if (this.props.open && event.key === "Escape") {
             event.preventDefault();
             event.stopImmediatePropagation();
-            this.props.onDropdownChanged(false);
+            this.props.onOpenChanged(false);
         }
     };
 
-    updatePosition = () => {
-        const {openerNode} = this.props;
-        if (openerNode) {
-            //$FlowFixMe typing is weird for the return value of ref callback
-            const rect = openerNode.getBoundingClientRect();
-            this.openerBounds = {
+    getOpenerBounds() {
+        const {openerElement} = this.props;
+        if (openerElement) {
+            const rect = openerElement.getBoundingClientRect();
+            return {
                 top: rect.top + window.pageYOffset,
                 left: rect.left + window.pageXOffset,
                 width: rect.width,
             };
         }
-    };
+    }
 
-    findTranslatedPosition() {
-        if (this.openerBounds) {
-            const {top, left, width} = this.openerBounds;
-            if (this.props.alignment === "left") {
-                return {
-                    transform: `translate3d(${left}px, ${top}px, 0)`,
-                };
-            } else {
-                return {
-                    transform: `translate3d(${left +
-                        width -
-                        (this.menuWidth || 0)}px, ${top}px, 0)`,
-                };
-            }
+    getTranslatedPosition(openerBounds: {
+        top: number,
+        left: number,
+        width: number,
+    }) {
+        const {top, left, width} = openerBounds;
+        if (this.props.alignment === "left") {
+            return {
+                transform: `translate3d(${left}px, ${top}px, 0)`,
+            };
+        } else {
+            return {
+                transform: `translate3d(${left +
+                    width -
+                    (this.menuWidth || 0)}px, ${top}px, 0)`,
+            };
         }
     }
 
     maybeRenderMenu() {
         const {items, light, open, style} = this.props;
+        const body = document.getElementsByTagName("body");
```

Comment:
If you use `document.querySelector("body")` here you won't have to do `body[0]` below.